linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -3304,6 +3320,16 @@ private object FunctionEmitter {
 
   private val thisOriginalName: OriginalName = OriginalName("this")
 
+  private val MaybeHijackedPrimNumberClasses = {
+    /* In theory, we could figure out the ancestors from the global knowledge,
+     * but that would be overkill.
+     */
+    Set(BoxedByteClass, BoxedShortClass, BoxedIntegerClass, BoxedFloatClass,
+        BoxedDoubleClass, ObjectClass, ClassName("java.lang.CharSequence"),
+        ClassName("java.io.Serializable"), ClassName("java.lang.Comparable"),
+        ClassName("java.lang.Number"))
+  }
```

Comment:
Can't we use `globalKnowledge.hijackedDescendants` (with set intersection or explicit contains checks) instead of listing supertypes here? (maybe in a follow-up commit)?

The original comment from the optimizer feels a bit misleading, since we're doing precisely what it describes as overkill in the global knowledge in the emitter :)

Reply:
Ah yes, good point. Indeed the emitter actually does have that in its global knowledged.

Updated.